File: macOs/Sources/DASCRM/Views/Reports/ReportsView.swift
"""
ReportsView.swift — DAS CRM macOS
Analytics Dashboard with Charts, Metrics, and KPI Tracking
Feature parity with Android ReportsAnalyticsScreen.tsx
"""

import SwiftUI

struct MetricData: Identifiable {
    let id = UUID()
    let label: String
    let value: String
    let trend: String
    let percentChange: String
    let icon: String
}

struct SalesMetric: Identifiable {
    let id = UUID()
    let month: String
    let revenue: Int
    let deals: Int
    let avgDealValue: Int
}

let fallbackMetrics = [
    MetricData(label: "Monthly Recurring Revenue", value: "₹24,50,000", trend: "UP",
              percentChange: "+12.5%", icon: "💰"),
    MetricData(label: "Sales Pipeline Value", value: "₹65,32,000", trend: "UP",
              percentChange: "+8.3%", icon: "🎯"),
    MetricData(label: "Customer Acquisition Cost", value: "₹15,200", trend: "DOWN",
              percentChange: "-3.2%", icon: "📊"),
    MetricData(label: "Average Deal Cycle", value: "28 days", trend: "DOWN",
              percentChange: "-5 days", icon: "⏱️"),
]

let fallbackSalesData = [
    SalesMetric(month: "January", revenue: 18500000, deals: 24, avgDealValue: 770833),
    SalesMetric(month: "February", revenue: 19200000, deals: 26, avgDealValue: 738461),
    SalesMetric(month: "March", revenue: 21800000, deals: 32, avgDealValue: 681250),
    SalesMetric(month: "April", revenue: 20500000, deals: 28, avgDealValue: 732142),
    SalesMetric(month: "May", revenue: 22100000, deals: 30, avgDealValue: 736666),
    SalesMetric(month: "June", revenue: 24500000, deals: 35, avgDealValue: 700000),
]

@MainActor
class ReportsViewModel: ObservableObject {
    @Published var salesData: [SalesMetric] = fallbackSalesData
    @Published var startDate = Date(timeIntervalSince1970: 1704067200) // 2024-01-01
    @Published var endDate = Date()
    @Published var chartType = "Line Chart"

    var maxRevenue: Int {
        salesData.map { $0.revenue }.max() ?? 0
    }

    var totalRevenue: Int {
        salesData.reduce(0) { $0 + $1.revenue }
    }
}

struct ReportsView: View {
    @StateObject private var viewModel = ReportsViewModel()
    @State private var showExportAlert = false

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Header
            VStack(alignment: .leading, spacing: 12) {
                Text("📊 Reports & Analytics")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(.white)

                // Date Range Selector
                HStack(spacing: 8) {
                    Text("📅 Date Range:")
                        .font(.system(size: 10))
                        .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))

                    DatePicker("", selection: $viewModel.startDate, displayedComponents: .date)
                        .datePickerStyle(.compact)
                        .frame(maxWidth: 150)

                    Text("to")
                        .font(.system(size: 10))
                        .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))

                    DatePicker("", selection: $viewModel.endDate, displayedComponents: .date)
                        .datePickerStyle(.compact)
                        .frame(maxWidth: 150)

                    Button(action: { showExportAlert = true }) {
                        Text("📥 Export Report")
                            .font(.system(size: 9, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 8)
                            .padding(.vertical, 4)
                            .background(Color(red: 0.31, green: 0.27, blue: 0.90))
                            .cornerRadius(4)
                    }
                    .buttonStyle(.plain)

                    Spacer()
                }
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.09, blue: 0.16))
            .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)

            // Content
            ScrollView {
                VStack(alignment: .leading, spacing: 12) {
                    // Key Metrics Grid (2x2)
                    VStack(alignment: .leading, spacing: 12) {
                        HStack(spacing: 12) {
                            MetricCardView(metric: fallbackMetrics[0])
                            MetricCardView(metric: fallbackMetrics[1])
                        }

                        HStack(spacing: 12) {
                            MetricCardView(metric: fallbackMetrics[2])
                            MetricCardView(metric: fallbackMetrics[3])
                        }
                    }

                    // Sales Performance Section
                    VStack(alignment: .leading, spacing: 12) {
                        HStack {
                            Text("💹 Sales Performance (Last 6 Months)")
                                .font(.system(size: 12, weight: .bold))
                                .foregroundColor(.white)

                            Spacer()

                            Picker("Chart Type", selection: $viewModel.chartType) {
                                Text("Line Chart").tag("Line Chart")
                                Text("Bar Chart").tag("Bar Chart")
                                Text("Area Chart").tag("Area Chart")
                            }
                            .pickerStyle(.menu)
                            .frame(width: 150)
                        }

                        // Chart Visualization
                        VStack(alignment: .center, spacing: 12) {
                            Text("📈 Chart Visualization")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)

                            Text("(Line chart: Revenue & Deal Count trend)")
                                .font(.system(size: 9))
                                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))

                            // Simple bar chart representation
                            HStack(alignment: .bottom, spacing: 6) {
                                ForEach(viewModel.salesData) { sale in
                                    VStack(spacing: 4) {
                                        RoundedRectangle(cornerRadius: 4)
                                            .fill(Color(red: 0.2, green: 0.83, blue: 0.60))
                                            .frame(width: 20, height: CGFloat(sale.revenue) / CGFloat(viewModel.maxRevenue) * 100)

                                        Text(sale.month.prefix(3))
                                            .font(.system(size: 8, weight: .bold))
                                            .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                                    }
                                }
                            }
                            .frame(height: 120)
                            .padding(.vertical, 8)

                            HStack {
                                Text("0")
                                    .font(.system(size: 8))
                                    .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                                Spacer()
                                Text("₹\(String(format: "%,d", viewModel.maxRevenue))")
                                    .font(.system(size: 8))
                                    .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                            }
                        }
                        .frame(maxWidth: .infinity, minHeight: 180)
                        .padding(12)
                        .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                        .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                        .cornerRadius(12)
                    }

                    // Sales Data Table
                    VStack(alignment: .leading, spacing: 12) {
                        Text("📋 Monthly Sales Data")
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(.white)

                        Table(viewModel.salesData) {
                            TableColumn("Month", value: \.month)
                            TableColumn("Revenue") { sale in
                                Text("₹\(String(format: "%,d", sale.revenue))")
                                    .foregroundColor(Color(red: 0.2, green: 0.83, blue: 0.60))
                            }
                            TableColumn("Deals") { sale in
                                Text(String(sale.deals))
                                    .foregroundColor(Color(red: 0.38, green: 0.65, blue: 0.98))
                            }
                            TableColumn("Avg Deal Value") { sale in
                                Text("₹\(String(format: "%,d", sale.avgDealValue))")
                                    .foregroundColor(Color(red: 0.98, green: 0.75, blue: 0.14))
                            }
                        }
                        .background(Color(red: 0.03, green: 0.04, blue: 0.07))
                        .onDoubleClickSelectAll(false)
                    }
                }
                .padding(16)
            }
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        }
        .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        .navigationTitle("Reports")
        .alert("📥 Export Report", isPresented: $showExportAlert) {
            Button("OK") { }
        } message: {
            Text("Exporting analytics report...")
        }
    }
}

struct MetricCardView: View {
    let metric: MetricData

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Text(metric.icon)
                    .font(.system(size: 16))

                VStack(alignment: .leading, spacing: 2) {
                    Text(metric.label)
                        .font(.system(size: 9))
                        .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                }

                Spacer()

                Text(metric.trend == "UP" ? "📈" : metric.trend == "DOWN" ? "📉" : "➡️")
                    .font(.system(size: 12))
            }

            HStack {
                Text(metric.value)
                    .font(.system(size: 16, weight: .bold))
                    .foregroundColor(Color(red: 0.2, green: 0.83, blue: 0.60))

                Text(metric.percentChange)
                    .font(.system(size: 10, weight: .bold))
                    .foregroundColor(
                        metric.trend == "UP" ? Color(red: 0.2, green: 0.83, blue: 0.60) :
                        metric.trend == "DOWN" ? Color(red: 0.98, green: 0.30, blue: 0.40) :
                        Color(red: 0.98, green: 0.75, blue: 0.14)
                    )

                Spacer()
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(12)
        .background(Color(red: 0.07, green: 0.09, blue: 0.16))
        .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
        .cornerRadius(12)
    }
}

#Preview {
    NavigationView {
        ReportsView()
    }
}

File: Win/ui/views/reports_view.py
"""
ReportsView.py — DAS CRM Windows
Analytics Dashboard with Charts, Metrics, and KPI Tracking
Feature parity with Android ReportsAnalyticsScreen.tsx
"""

from PyQt6.QtWidgets import (
    QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QScrollArea,
    QFrame, QGridLayout, QComboBox, QDateEdit, QAbstractItemView, QTableWidget,
    QTableWidgetItem
)
from PyQt6.QtCore import Qt, QDate
from PyQt6.QtGui import QFont, QBrush, QColor
from dataclasses import dataclass

# ─────────────────────────────────────────────────────────────────────────────────────
# DATA MODELS
# ─────────────────────────────────────────────────────────────────────────────────────

@dataclass
class MetricData:
    """Analytics metric"""
    label: str
    value: str
    trend: str  # UP, DOWN, STABLE
    percentChange: str
    icon: str

@dataclass
class SalesMetric:
    """Sales performance metric"""
    month: str
    revenue: int
    deals: int
    avgDealValue: int

FALLBACK_METRICS = [
    MetricData("Monthly Recurring Revenue", "₹24,50,000", "UP", "+12.5%", "💰"),
    MetricData("Sales Pipeline Value", "₹65,32,000", "UP", "+8.3%", "🎯"),
    MetricData("Customer Acquisition Cost", "₹15,200", "DOWN", "-3.2%", "📊"),
    MetricData("Average Deal Cycle", "28 days", "DOWN", "-5 days", "⏱️"),
]

FALLBACK_SALES_DATA = [
    SalesMetric("January", 18500000, 24, 770833),
    SalesMetric("February", 19200000, 26, 738461),
    SalesMetric("March", 21800000, 32, 681250),
    SalesMetric("April", 20500000, 28, 732142),
    SalesMetric("May", 22100000, 30, 736666),
    SalesMetric("June", 24500000, 35, 700000),
]

# ─────────────────────────────────────────────────────────────────────────────────────
# METRIC CARD COMPONENT
# ─────────────────────────────────────────────────────────────────────────────────────

class MetricCard(QFrame):
    """Analytics metric display card"""
    def __init__(self, metric: MetricData, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 16px;
            }
        """)

        layout = QVBoxLayout(self)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(8)

        # Header
        headerLayout = QHBoxLayout()
        iconLabel = QLabel(metric.icon)
        iconLabel.setFont(QFont("Segoe UI", 18))
        headerLayout.addWidget(iconLabel)

        labelWidget = QLabel(metric.label)
        labelWidget.setFont(QFont("Segoe UI", 9))
        labelWidget.setStyleSheet("color: #94a3b8;")
        headerLayout.addWidget(labelWidget)

        # Trend
        trendEmoji = "📈" if metric.trend == "UP" else "📉" if metric.trend == "DOWN" else "➡️"
        trendLabel = QLabel(trendEmoji)
        trendLabel.setFont(QFont("Segoe UI", 12))
        headerLayout.addStretch()
        headerLayout.addWidget(trendLabel)

        layout.addLayout(headerLayout)

        # Value and change
        valueLayout = QHBoxLayout()
        valueLabel = QLabel(metric.value)
        valueLabel.setFont(QFont("Segoe UI", 16, QFont.Weight.Bold))
        valueLabel.setStyleSheet("color: #34d399;")
        valueLayout.addWidget(valueLabel)

        changeColor = "#34d399" if metric.trend == "UP" else "#ef4444" if metric.trend == "DOWN" else "#fbbf24"
        changeLabel = QLabel(metric.percentChange)
        changeLabel.setFont(QFont("Segoe UI", 10, QFont.Weight.Bold))
        changeLabel.setStyleSheet(f"color: {changeColor};")
        valueLayout.addWidget(changeLabel)
        valueLayout.addStretch()

        layout.addLayout(valueLayout)

# ─────────────────────────────────────────────────────────────────────────────────────
# MAIN REPORTS VIEW
# ─────────────────────────────────────────────────────────────────────────────────────

class ReportsView(QWidget):
    """Analytics & Reports Dashboard"""

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QWidget { background-color: #090d16; }
            QLabel { color: #f8fafc; }
            QPushButton { color: white; font-weight: bold; border: none; border-radius: 6px; }
        """)

        self.salesData = list(FALLBACK_SALES_DATA)
        self._build_ui()

    def _build_ui(self):
        """Build reports UI"""
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)

        scrollArea = QScrollArea()
        scrollArea.setWidgetResizable(True)
        scrollArea.setStyleSheet("QScrollArea { border: none; background-color: #090d16; }")

        scrollWidget = QWidget()
        scrollLayout = QVBoxLayout(scrollWidget)
        scrollLayout.setContentsMargins(16, 16, 16, 24)
        scrollLayout.setSpacing(12)

        # Title
        titleLabel = QLabel("📊 Reports & Analytics")
        titleLabel.setFont(QFont("Segoe UI", 14, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        scrollLayout.addWidget(titleLabel)

        # Date Range Selector
        dateLayout = QHBoxLayout()
        dateLayout.setContentsMargins(0, 0, 0, 0)
        dateLayout.setSpacing(8)

        dateLabel = QLabel("📅 Date Range:")
        dateLabel.setFont(QFont("Segoe UI", 10))
        dateLabel.setStyleSheet("color: #cbd5e1;")
        dateLayout.addWidget(dateLabel)

        self.startDateInput = QDateEdit()
        self.startDateInput.setDate(QDate(2026, 1, 1))
        self.startDateInput.setStyleSheet("""
            QDateEdit {
                background-color: #0f172a;
                color: #ffffff;
                border: 1px solid #1e293b;
                border-radius: 6px;
                padding: 6px;
            }
        """)
        dateLayout.addWidget(self.startDateInput)

        toLabel = QLabel("to")
        toLabel.setFont(QFont("Segoe UI", 10))
        toLabel.setStyleSheet("color: #cbd5e1;")
        dateLayout.addWidget(toLabel)

        self.endDateInput = QDateEdit()
        self.endDateInput.setDate(QDate.currentDate())
        self.endDateInput.setStyleSheet("""
            QDateEdit {
                background-color: #0f172a;
                color: #ffffff;
                border: 1px solid #1e293b;
                border-radius: 6px;
                padding: 6px;
            }
        """)
        dateLayout.addWidget(self.endDateInput)

        btnExport = QPushButton("📥 Export Report")
        btnExport.setStyleSheet("background-color: #4f46e5; padding: 4px 8px; font-size: 9px;")
        btnExport.clicked.connect(lambda: None)
        dateLayout.addWidget(btnExport)

        dateLayout.addStretch()
        scrollLayout.addLayout(dateLayout)

        # Key Metrics Grid (2x2)
        metricsLayout = QGridLayout()
        metricsLayout.setContentsMargins(0, 0, 0, 0)
        metricsLayout.setSpacing(12)

        for idx, metric in enumerate(FALLBACK_METRICS):
            card = MetricCard(metric)
            metricsLayout.addWidget(card, idx // 2, idx % 2, 1, 1)

        scrollLayout.addLayout(metricsLayout)

        # Sales Performance Section
        salesTitleLayout = QHBoxLayout()
        salesTitleLabel = QLabel("💹 Sales Performance (Last 6 Months)")
        salesTitleLabel.setFont(QFont("Segoe UI", 12, QFont.Weight.Bold))
        salesTitleLabel.setStyleSheet("color: #ffffff;")
        salesTitleLayout.addWidget(salesTitleLabel)

        chartTypeCombo = QComboBox()
        chartTypeCombo.addItems(["Line Chart", "Bar Chart", "Area Chart"])
        chartTypeCombo.setStyleSheet("""
            QComboBox {
                background-color: #0f172a;
                color: #ffffff;
                border: 1px solid #1e293b;
                border-radius: 6px;
                padding: 4px;
            }
        """)
        salesTitleLayout.addStretch()
        salesTitleLayout.addWidget(chartTypeCombo)

        scrollLayout.addLayout(salesTitleLayout)

        # Chart Placeholder
        chartFrame = QFrame()
        chartFrame.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
            }
        """)
        chartLayout = QVBoxLayout(chartFrame)
        chartLayout.setContentsMargins(16, 16, 16, 16)

        chartLabel = QLabel("📈 Chart Visualization\n(Line chart: Revenue & Deal Count trend)")
        chartLabel.setFont(QFont("Segoe UI", 10))
        chartLabel.setStyleSheet("color: #94a3b8;")
        chartLabel.setAlignment(Qt.AlignmentFlag.AlignCenter)
        chartLayout.addWidget(chartLabel)

        scrollLayout.addWidget(chartFrame, 1)

        # Sales Data Table
        tableTitleLabel = QLabel("📋 Monthly Sales Data")
        tableTitleLabel.setFont(QFont("Segoe UI", 12, QFont.Weight.Bold))
        tableTitleLabel.setStyleSheet("color: #ffffff;")
        scrollLayout.addWidget(tableTitleLabel)

        self.salesTable = QTableWidget()
        self.salesTable.setColumnCount(4)
        self.salesTable.setHorizontalHeaderLabels(["Month", "Revenue", "Deals", "Avg Deal Value"])
        self.salesTable.horizontalHeader().setStretchLastSection(False)
        self.salesTable.setSelectionBehavior(QAbstractItemView.SelectionBehavior.SelectRows)
        self.salesTable.setSelectionMode(QAbstractItemView.SelectionMode.SingleSelection)
        self.salesTable.setStyleSheet("""
            QTableWidget { background-color: #030712; gridline-color: #1e293b; }
            QHeaderView::section { background-color: #0b1329; color: #818cf8; padding: 6px;
                                   border: none; border-right: 1px solid #1e293b; }
        """)

        self.salesTable.setColumnWidth(0, 120)
        self.salesTable.setColumnWidth(1, 150)
        self.salesTable.setColumnWidth(2, 100)
        self.salesTable.setColumnWidth(3, 150)

        self._refresh_sales_table()

        scrollLayout.addWidget(self.salesTable)

        scrollArea.setWidget(scrollWidget)
        layout.addWidget(scrollArea)

    def _refresh_sales_table(self):
        """Refresh sales data table"""
        self.salesTable.setRowCount(len(self.salesData))

        for rowIdx, sale in enumerate(self.salesData):
            self.salesTable.setItem(rowIdx, 0, QTableWidgetItem(sale.month))

            revenueItem = QTableWidgetItem(f"₹{sale.revenue:,}")
            revenueItem.setForeground(QBrush(QColor("#34d399")))
            self.salesTable.setItem(rowIdx, 1, revenueItem)

            dealsItem = QTableWidgetItem(str(sale.deals))
            dealsItem.setForeground(QBrush(QColor("#60a5fa")))
            self.salesTable.setItem(rowIdx, 2, dealsItem)

            avgItem = QTableWidgetItem(f"₹{sale.avgDealValue:,}")
            avgItem.setForeground(QBrush(QColor("#fbbf24")))
            self.salesTable.setItem(rowIdx, 3, avgItem)
